m/processing_tools/gridGliderData.m

```diff
@@ -306,26 +306,31 @@
   fprintf('  number of variables   : %d\n', num_variables);
   data_grid_variables = nan(num_casts, num_levels, num_variables);
   for cast_idx = 1:num_casts
-    cast_select = (profile == cast_idx);
-    cast_lat = latitude(cast_select);
-    cast_lon = longitude(cast_select);
-    cast_depth = depth(cast_select);
-    cast_time = time(cast_select);
-    cast_variables = variables(cast_select, :);
-    data_grid.time(cast_idx) = nanmean(cast_time);
-    data_grid.latitude(cast_idx) = nanmean(cast_lat);
-    data_grid.longitude(cast_idx) = nanmean(cast_lon);
-    if ~isempty(cast_variables) % Speed up when there are no variables.
-      data_grid_variables(cast_idx, :, :) = ...
-        cell2mat(arrayfun(@(d) nanmean(cast_variables(abs(cast_depth-d)<=0.5*depth_resolution, :), 1), ...
-                          depth_range(:), 'UniformOutput', false));
-    end
-  end
-  % Move binned variable data to output struct.
-  for variable_name_idx = 1:num_variables
-    variable_name = variable_name_list{variable_name_idx};
-    data_grid.(variable_name) = data_grid_variables(:, :, variable_name_idx);
-  end
+      cast_select = (profile == cast_idx);
+      cast_lat = latitude(cast_select);
+      cast_lon = longitude(cast_select);
+      cast_depth = depth(cast_select);
+      cast_time = time(cast_select);
+      cast_variables = variables(cast_select, :);
+      data_grid.time(cast_idx) = nanmean(cast_time);
+      data_grid.latitude(cast_idx) = nanmean(cast_lat);
+      data_grid.longitude(cast_idx) = nanmean(cast_lon);
+
+      % Speed up when there are no variables. % -FC modifs
+      if ~isempty(cast_variables)
+          cast_mean = arrayfun(@(d) nanmean(cast_variables(abs(cast_depth-d)<=0.5*depth_resolution,:),1), depth_range(:), 'UniformOutput', false);
+          
+          cast_good_idx = cellfun(@(d) any(~isnan(d(:))), cast_mean);     
+          cast_mean = cell2mat(cast_mean(cast_good_idx));
+          data_grid_variables(cast_idx, cast_good_idx, :) =  cast_mean;
+      end
+   end
+  
+   % Move binned variable data to output struct.
+   for variable_name_idx = 1:num_variables
+       variable_name = variable_name_list{variable_name_idx};
+       data_grid.(variable_name) = data_grid_variables(:, :, variable_name_idx);
+   end
```

Comment:
This is the source of the error, that should return a row matrix of 1 x 3 NaNs,
to be compatible with what `nanmean` returns in Octave (from forge package statistics)
and MATLAB (from Statistics Toolbox). In fact, their implementations are almost identical,
and you can satisfy the dependency by borrowing the Octave's [implementation][1]
and placing it somewhere in your path (e.g. in a private folder), with minimal syntax changes:
```matlab
function M = nanmean (X, dim)
  nans = isnan(X);
  X(nans) = 0;
  if nargin == 1
    n = sum(~nans);
    n(n == 0) = NaN;
    M = sum(X) ./ n;
  else
    n = sum(~nans, dim);
    n(n == 0) = NaN;
    M = sum(X, dim) ./ n;
  end
end
```

  [1]: https://sourceforge.net/p/octave/statistics/ci/default/tree/inst/nanmean.m